governance/policies/archiving.md

```diff
@@ -0,0 +1,102 @@
+  - If a new maintainer steps forward, they will have 1 week to fix the package version.
+  - Otherwise, the package will be archived.
+
+
+### Archiving a package
+
+When it has been decided that a set of package versions (aka "versions") should be archived, archiving will proceed according to the following process:
+
+- A PR will be made to add the archived versions to the archive repo. 
+    - The opam file of each version will be extended as follows:
+        - Any dependencies without upper bounds will have upper bounds added, recording the latest available version of the dependency in the primary repo at the time of archiving.
+        -  The field `x-reason-for-archiving` will be added.
+        -  The field `x-opam-repository-commit-hash-at-time-of-archiving` will be added.
+- A PR will be made to remove the versions from the primary repo:
+  - The removal PR should link to the corresponding archiving PR.
+  - The commit message should have the title `Archive packages` and its body should contain the hash of the commit that adds the packages to the archive.
+- A announcement will be made on discuss.ocaml.org
+- After waiting 1 week for feedback, the PRs will be merged.
+
+### Specification of the `x-` fields used in the archiving process
+
+- `x-reason-for-archiving`:
+    - Allowed values: a list of containing one more of the following strings
+      `ocaml-version`, `source-unavailable`, `maintenance-intent`, or
+      `uninstallable`.
+    - Meaning: Records the unmet [primary repo criteria](#inclusion-criteria)
+      motivating the archiving, as follows:
+        1. `ocaml-version`: The package no longer met the compiler cutoff threshold.
+        2. `source-unavailable`: The sources of the package version became unavailable.
+        3. `maintenance-intent`: The package version falls outside of a
+           package's maintenance intent, and it is not a dependency of a package
+           satisfying any of the primary repo criteria.
+        4. `uninstallable`: The package version is not installable on any
+           of the supported platforms.
+    - Example: `["ocaml-version", "source-unavailable"]`
+- `x-opam-repository-commit-hash-at-time-of-archiving`:
+    - Allowed values: a string
+    - Meaning: Records the commit hash of the primary repo at the time a package version is archived.
+    - Example: `"ca32ab3b976aa7abc00c7605548f78a30980d35b"`
+- `x-maintenance-intent`:
+    - Allowed values: a list of strings matching version numbers, possibly using the special keywords `(latest)`, `(any)` and `(none)`
+    - Meaning:
+        - Expresses the declared intent of the maintainers to maintain only certain versions ranges of a package.
+        - The value of the `x-maintenance-intent` on the latest published package will precedence.
+    - Examples:
+        - `["(latest)"]` the maintainer will only maintain the latest version
```

Comment:
An aside: do we have an OCaml implementation of this, similar to `ocaml-version`?

Reply:
No, not yet. We will be working on that after everything needed for the first two phases, which are independent of this, is ready